### PR TITLE
test: Setup project example

### PR DESCRIPTION
**Pull request by dtakahas**
@jrtcppv Please ignore the docstring additions. I started on them and then decided to save it for another PR.

I've commented 🍖  on the meaty, important bits.  You can run the example with the instructions in [examples/README.md](https://github.com/cvisionai/tator-r/blob/dave/setup-project-example/examples/README.md)

### CODE REVIEW

File: R/utilities.R [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+#'  @param host URL of host. Default is https://www.tatorapp.com.
+#'  @param token API token.
+#'  @return TatorApi object
+#' @export
+get_api <- function(host = "https://www.tatorapp.com", token = Sys.getenv("TATOR_TOKEN")) {
+  instance <- TatorApi$new()
+  instance$apiClient$apiKeys["Authorization"] <- stringr::str_interp("Token ${token}")
+  instance$apiClient$basePath <- host
+  return(instance)
+}
```
Comment (dtakahas) [inline R/utilities.R:12]: 🍖

File: examples/setup_project.R [hunk 2]
```diff
@@ -0,0 +1,356 @@
+#!/usr/bin/env Rscript
+
+# This example demonstrates how to create a project and configure media
+# and metadata type definitions.
```
Comment (dtakahas) [inline examples/setup_project.R:4]: 🥩

**Review by jrtcppv — COMMENTED** [review]
The instructions actually failed while following instructions in the README:
```
$ Rscript examples/setup_project.R --host="https://xxx.tator.dev" --token="xxx"
Error in library(optparse) : there is no package called 'optparse'
Execution halted
```
Can we install dependencies in `build.sh`?

**Comment by dtakahas** [comment]
Ah that package is only used in the example, so that's why the build script
didn't grab it. I think the appropriate thing to do is add optparse to the
'Suggests' section of the dependency manifest. I'll check that out.

**Comment by dtakahas** [comment]
@jrtcppv The build script has been improved. Anyone should be able to start from even a fresh install of R and use the shell script. Can you give it another try? Same steps, the build script should be additive / idempotent. However, since the example has a hard coded project name, running that will fail on a second run unless you change it up. Thanks.

EDIT: I'll address the build versioning later when we get to submitting to CRAN

**Comment by jrtcppv** [comment]
Now I get 
```
$ ./build.sh
Error: unexpected '/' in "if (!require(devtools, character.only = TRUE)) {install.packages(devtools, repos=http:/"
Execution halted
```

I am doing this in the R-studio shell that comes with windows, is that a valid environment? My guess is most of our users will be on windows.

**Comment by jrtcppv** [comment]
It's probably a MinGW shell if I had to guess, if that helps.

**Pull request merged**